code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -579,6 +678,60 @@
 		M.Jitter(2)
 	..()
 
+/datum/reagent/medicine/morphine/addiction_act_stage2(mob/living/M)
+	if(prob(33))
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+		M.adjustToxLoss(1*REM, 0)
+		. = 1
+		M.Dizzy(3)
+		M.Jitter(3)
+	..()
+
+/datum/reagent/medicine/morphine/addiction_act_stage3(mob/living/M)
+	if(prob(33))
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+		M.adjustToxLoss(2*REM, 0)
+		. = 1
+		M.Dizzy(4)
+		M.Jitter(4)
+	..()
+
+/datum/reagent/medicine/morphine/addiction_act_stage4(mob/living/M)
+	if(prob(33))
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+		M.adjustToxLoss(3*REM, 0)
+		. = 1
+		M.Dizzy(5)
+		M.Jitter(5)
+	..()
+
+/datum/reagent/medicine/ether
+	name = "Ether"
+	id = "ether"
+	description = "A strong but highly addictive anesthetic and sedative."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	addiction_threshold = 25
+
+/datum/reagent/medicine/morphine/on_mob_life(mob/living/M)
```

Comment:
I think this should be /datum/reagent/medicine/ether/on_mob_life(mob/living/M)
Also, the next addiction_act functions